.github/workflows/publish.yml

```diff
@@ -61,19 +98,25 @@ jobs:
       ##########################################################################
       - name: ENV - VERSION
         run: echo "VERSION=$(node -p -e "require('./package.json').version")" >> $GITHUB_ENV
+      - name: ENV - BUILD_NUMBER
+        run: echo "BUILD_NUMBER=$(node -p -e "require('./package.json').build-number")" >> $GITHUB_ENV
+      - name: ENV - DOCKER_ORGANISATION_SOURCE
+        run: echo "DOCKER_ORGANISATION_SOURCE=$(node -p -e "require('./package.json').docker-organisation-source")" >> $GITHUB_ENV
+      - name: ENV - DOCKER_ORGANISATION_PUSH
+        run: echo "DOCKER_ORGANISATION_PUSH=$(node -p -e "require('./package.json').docker-organisation-push")" >> $GITHUB_ENV
       - name: ENV - BUILD_DATE
         run: echo "BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_ENV
       - name: ENV - BUILD_VERSION
-        run: echo "BUILD_VERSION=${VERSION}.${GITHUB_RUN_NUMBER}" >> $GITHUB_ENV
+        run: echo "BUILD_VERSION=${VERSION}-${BUILD_NUMBER}" >> $GITHUB_ENV
       - name: ENV - BUILD_COMMIT
         run: echo "BUILD_COMMIT=${GITHUB_SHA}" >> $GITHUB_ENV
       ##########################################################################
       # BUILD WEBAPP DOCKER IMAGE (build) ######################################
       ##########################################################################
       - name: webapp | Build `branded` image
         run: |
-          docker build --target branded -t "ocelotsocialnetwork/webapp-branded:latest" -t "ocelotsocialnetwork/webapp-branded:${VERSION}" -t "ocelotsocialnetwork/webapp-branded:${BUILD_VERSION}" -f docker/webapp.Dockerfile .
-          docker save "ocelotsocialnetwork/webapp-branded" > /tmp/webapp-branded.tar
+          docker build --target branded -t "${DOCKER_ORGANISATION_PUSH}/webapp-branded:latest" -t "${DOCKER_ORGANISATION_PUSH}/webapp-branded:${VERSION}" -t "${DOCKER_ORGANISATION_PUSH}/webapp-branded:${BUILD_VERSION}" -f docker/webapp.Dockerfile --build-arg APP_IMAGE=${DOCKER_ORGANISATION_SOURCE}/webapp:${BUILD_VERSION} .
+          docker save "${DOCKER_ORGANISATION_PUSH}/webapp-branded" > /tmp/webapp-branded.tar
```

Comment:
`APP_IMAGE=${DOCKER_ORGANISATION_SOURCE}/webapp:${BUILD_VERSION}`

This is not easy to read.

I beleive we can keep `${DOCKER_ORGANISATION_SOURCE}` as static string `ocelotsocialnetwork` since its very unlikely it would change and just confuses the reader.

The `${BUILD_VERSION}` part is wrong tho, since this must be defined specifically. Possibly in the `package.json` with a value like `ocelot-version: "1.0.2"` or similar.

If you wanna go real fancy you can run a script which does somehting like this:
- check all available tags on github for ocelot
- let the user decide which one he wanna use
- update package.json accordingly
- increase package.json version
- ...
- profit

Reply:
Okay. I was on the path that we use the `package.json` `version` directly for it.

- I removed the `DOCKER_ORGANISATION_SOURCE` and made it static again.
- I implement `ocelot-docker-version-tag`.
- And I implement a `BUILD_VERSION` with `version`, `GITHUB_RUN_NUMBER`, and `ocelot-docker-version-tag` oriented at the NodeJS docker tag: `node:12.19.0-alpine3.10`, though ours looks like `app-branded:1.0.2-3-ocelot.social1.0.2-79`.

I’ve put the script idea in a separate issue:
https://github.com/Ocelot-Social-Community/Ocelot-Social-Deploy-Rebranding/issues/9